src/ROUTES.js

```diff
@@ -78,22 +75,21 @@ export default {
     IOU_REQUEST,
     IOU_BILL,
     IOU_SEND,
-    IOU_REQUEST_WITH_REPORT_ID: `${IOU_REQUEST}/:reportID?`,
-    IOU_BILL_WITH_REPORT_ID: `${IOU_BILL}/:reportID?`,
-    IOU_SEND_WITH_REPORT_ID: `${IOU_SEND}/:reportID?`,
-    getIouRequestRoute: (reportID) => `${IOU_REQUEST}/${reportID}`,
-    getIouSplitRoute: (reportID) => `${IOU_BILL}/${reportID}`,
-    getIOUSendRoute: (reportID) => `${IOU_SEND}/${reportID}`,
-    IOU_BILL_CURRENCY: `${IOU_BILL_CURRENCY}/:reportID?`,
-    IOU_REQUEST_CURRENCY: `${IOU_REQUEST_CURRENCY}/:reportID?`,
-    MONEY_REQUEST_DESCRIPTION: `${IOU_REQUEST}/description`,
-    IOU_SEND_CURRENCY: `${IOU_SEND_CURRENCY}/:reportID?`,
+    MONEY_REQUEST: ':iouType/new/:reportID?',
+    MONEY_REQUEST_AMOUNT: ':iouType/new/amount/:reportID?',
+    MONEY_REQUEST_PARTICIPANTS: ':iouType/new/participants/:reportID?',
+    MONEY_REQUEST_CONFIRMATION: ':iouType/new/confirmation/:reportID?',
+    MONEY_REQUEST_CURRENCY: ':iouType/new/currency/:reportID?',
+    MONEY_REQUEST_DESCRIPTION: ':iouType/new/description/:reportID?',
```

Comment:
I'm not really sure about using a variable in the beginning of the url. This will match anything `/randomword/new/`.

@PauloGasparSv @thienlnam What are you thoughts on this?

Reply:
We will show not found page if it's not one of send, request, or split.

Reply:
If we are only showing for send/request/split seems fine to me - the alternative is to create a specific route for each which I don't like as much

Reply:
Where are these `iouType` being defined? I understand this is nicer than having a route for each, btu just looking at this I cant figure out where the :iouType is defined so I think we should add a comment about this so people can quickly know where new iouType needs to be added or what not

Reply:
Added comment